### PR TITLE
Auto-approve comments

### LINKED ISSUE #4262: Comments: replying to comment does not auto-approve
### Expected behavior
In the Comments section of the app, I send a reply to a pending comment. I expect this to approve the comment and post my reply because that is how it works from notifications and WP Admin.

### Actual behavior
My reply gets 'posted' but the comment I replied to is not automatically approved, so neither appears on my site.

### Steps to reproduce the behavior
1. Have a comment in moderation.
2. Open the Comments section of the app and find that comment.
3. Reply to that comment.

##### Tested on Nexus 6P, Android 6.0.1

Raised in Helpshift: 111856-h

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #4262 

To test:

1. Have a comment in moderation.
2. Open the Comments section of the app and find that comment.
3. Reply to that comment.
4. Verify the original comment is now approved.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java [hunk 1]
```diff
@@ -820,6 +820,9 @@ public void onActionResult(CommentActionResult result) {
                         ToastUtils.showToast(getActivity(), getString(R.string.note_reply_successful));
                         mEditReply.setText(null);
                         mEditReply.getAutoSaveTextHelper().clearSavedText(mEditReply);
+
+                        // approve the comment
+                        moderateComment(CommentStatus.APPROVED);
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java]: Shouldn't this be done only if the comment isn't already approved?

Comment (0nko) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java]: You're right. Do you think it's enough to just check for the approved status or should there be some more elaborate logic for other states? The thing is in wp-admin you can't reply to trashed or spam comments but you can on mobile.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java]: I think for this PR simply checking for the approved status is all that's needed. We may want a separate PR, though, to bring us in line with wp-admin.

Comment (0nko) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java]: I've made the changes. I'll create another issue for the wp-admin sync.

**Comment by nbradbury** [comment]
:shipit:

**Comment by 0nko** [comment]
Thanks Nick!

**Pull request merged**